### PR TITLE
Fix TP non "fixed-sized" mode

### PR DESCRIPTION
**Pull request by breznak**
Fixes: #1360  

DON'T merge until both Py, C++, TP10X2 are resolved!
c++ sibling: https://github.com/numenta/nupic.core/pull/484

- [ ] revert `.nupic_modules` pointing to my local SHA
- [x] python non fixed-sized TP works
- [ ] fix binding from c++ in TP10X2

### CODE REVIEW

File: .nupic_modules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Default nupic.core dependencies (override in optional .nupic_config)
-NUPIC_CORE_REMOTE = 'git://github.com/numenta/nupic.core.git'
-NUPIC_CORE_COMMITISH = '140a176ef1f53bb5014e60cec5a26ecc0498d23e'
+NUPIC_CORE_REMOTE = 'git://github.com/breznak/nupic.core.git' # FIXME revert before merge!
```
Comment (breznak) [inline .nupic_modules]: this needs to be reverted before final merge!

**Comment by breznak** [comment]
@oxtopus @scottpurdy @rhyolight please review. 

The Py part runs fine. C++ has error: 

```
Traceback (most recent call last):
  File "/home/mmm/devel/nupic/nupic-source/tests/unit/nupic/research/tp_test.py", line 106, in testCheckpointMiddleOfSequence
    tp1.compute(bottomUpInput, True, True)
  File "/home/mmm/devel/nupic/nupic-source/nupic/research/TP10X2.py", line 312, in compute
    y = self.cells4.compute(bottomUpInput, computeInfOutput, enableLearn)
  File "/home/mmm/devel/nupic/nupic-source/nupic/bindings/algorithms.py", line 2219, in compute
    return _algorithms.Cells4_compute(self, *args)
RuntimeError: ASSERTION FAILED: "segIdx < _segments.size()" 

```

**Comment by rhyolight** [comment]
Should we still do this? The temporal memory algorithm replaces this.

**Comment by breznak** [comment]
@rhyolight can we add 'wontfix' label on such closed PRs, so it's possible to revisit when needed? 

Actually, I have a proposal: can we merge the part "fixes python part of the issue", even if we don't have compatibility with c++ version (we don't have it anyway as it does not work), atleast it's working for python. 
Or we should evaluate the functionality (still useful?) and remove completely.

**Comment by rhyolight** [comment]
I would like to move ahead with TM and get rid of the TP as soon as we can. @scottpurdy What do you think about it?

**Comment by scottpurdy** [comment]
@rhyolight - Tons of work before we could consider that. AFAIK, TP is still better for time series problems.

**Comment by rhyolight** [comment]
Ok then, we should reopen this!

**Comment by rhyolight** [comment]
Closing due to lack of activity. Please re-open if progress is made.

**Pull request closed without merging**